pylintrc

```diff
@@ -30,7 +30,7 @@ max-line-length=110
 [DESIGN]
 max-args=10
 max-attributes=15
-max-public-methods=35
+max-public-methods=40
```

Comment:
This is possibly something that the project owners will want to look at, but IMO it's fine in this repo

Reply:
Although this seems to be reverted in the next commit?

Reply:
Yes so there were 2 approaches in which this could have done. Earlier I was merging the project config with the MR config and then taking a decision.

But then I thought that there is only one case we need to address which is when at project level we have `sqaush_option: always`, so we are not taking any decision on behalf of the user or gitlab, instead we are respecting user's configuration and only sending `squash: true` when the project level configuration is set to always because in that case even gitlab won't allow the user to change the configuration at MR level.

For the earlier case I had to add a method to `merge_request` class which was exceeding the number of methods, but in this case we don't have to deal with this class at all, hence reverted.